ServiceFabric.PubSubActors/SubscriberServices/SubscriberStatelessServiceBase.cs

```diff
@@ -62,12 +65,14 @@ protected override async Task OnOpenAsync(CancellationToken cancellationToken)
         /// Registers this service as a subscriber for the given message type.
         /// </summary>
         /// <param name="messageType">Full type of message object.</param>
-        public async Task SubscribeAsync(Type messageType)
+        public Task SubscribeAsync(Type messageType)
         {
-            if (messageType.FullName != null && _subscriptions[messageType.FullName] is SubscriptionDefinition subscription)
+            if (messageType.FullName != null && Subscriptions.TryGetValue(messageType.FullName, out var subscription))
             {
-                await _subscriberServiceHelper.RegisterMessageTypeAsync(this, messageType, subscription.Broker);
+                return _subscriberServiceHelper.RegisterMessageTypeAsync(this, messageType, subscription.Broker);
```

Comment:
I see why the async/await wasn't necessary.  We should update UnsubscribeAsync() as well.